### ISSUE TITLE
Enable `useDocument` option again

### ISSUE THREAD

**Issue by toomuchdesign** [toomuchdesign/next-page-tester#268]
## 🚀 Feature Proposal

`useDocument` option was disabled in #267 (v0.29.0) because of #263.

As mentioned in #263, we should try to enable it again as soon as Next.js document rendering APIs stabilize.

[Here](https://github.com/vercel/next.js/commit/08a24780b01608e16b7733d31bda2aaf36fd1750#diff-63632e6e580e69692941c89bab0a0e0436ade3f9543ef610925f795928d9d5fdL271) is how the previous `Document.renderDocument` call was replaced.

- Original `Document.renderDocument` (v11.1.0) [🔗](https://github.com/vercel/next.js/blob/v11.1.0/packages/next/pages/_document.tsx#L179)
- Original `Document.renderDocument` call (v11.1.0) [🔗](https://github.com/vercel/next.js/blob/v11.1.0/packages/next/server/render.tsx#L271)
- New document rendering implementation (v11.1.2) [🔗](https://github.com/vercel/next.js/blob/v11.1.2/packages/next/server/render.tsx#L1101)
- Newer canary document rendering implementation (v11.1.3-canary.30) [🔗](https://github.com/vercel/next.js/blob/v11.1.3-canary.30/packages/next/server/render.tsx#L1089)

**Comment by dbehmoaras** [comment]
If the useDocument feature is disable, does this mean we can't run tests on individual pages for UI specific features?

**Comment by toomuchdesign** [comment]
No, `useDocument` renders [Next.js custom document](https://nextjs.org/docs/advanced-features/custom-document).
